### PR TITLE
Spark Yarn Support

### PR DESCRIPTION
**Pull request by sdreyer**
# Description

Allows yarn configs to be uploaded and used in park

### Select type(s) of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

# Checklist:

- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have fixed any merge conflicts

### CODE REVIEW

File: client/tests/resources_test.py [hunk 1]
```diff
@@ -226,6 +227,71 @@ def bigquery_provider(bigquery_config):
...
+def test_with_paths(core_site_path, yarn_site_path):
+    config = {
+        "master": "yarn",
+        "deploy_mode": "client",
+        "python_version": "3.7.16",
+        "core_site_path": core_site_path,
+        "yarn_site_path": yarn_site_path,
+    }
+
+    assert SparkCredentials(**config)._verify_yarn_config() is None
```
Comment (ahmadnazeri) [inline client/tests/resources_test.py:249]: do we need the `is None`?

File: provider/spark.go [hunk 2]
```diff
@@ -684,10 +687,49 @@ func (s *SparkGenericExecutor) InitializeExecutor(store SparkFileStore) error {
 	return nil
 }
 
+func (s *SparkGenericExecutor) getYarnCommand(args string) (string, error) {
+	configDir, err := os.MkdirTemp("", "hadoop-conf")
+	if err != nil {
+		return "", fmt.Errorf("could not create temp dir: %v", err)
+	}
+	coreSitePath := filepath.Join(configDir, "core-site.xml")
+	yarnSitePath := filepath.Join(configDir, "yarn-site.xml")
```
Comment (ahmadnazeri) [inline provider/spark.go:696]: small thing but might be better to pair this with the write below.

**Review by aolfat — DISMISSED** [review]

**Review by ahmadnazeri — APPROVED** [review]

File: client/src/featureform/register.py [hunk 3]
```diff
@@ -1907,6 +1907,8 @@ def register_hdfs(
         path: str = "",
         description: str = "",
         team: str = "",
+        tags: List[str] = [],
```
[the remark this answers is not in the document]
Comment (sdreyer) [reply client/src/featureform/register.py:1910]: Yeah was just gonna let someone do them at once

File: provider/spark.go (hunk 2 above)
Comment (sdreyer) [reply provider/spark.go]: fixed

**Review by aolfat — APPROVED** [review]

**Pull request merged**